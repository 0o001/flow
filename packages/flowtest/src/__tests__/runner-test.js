/**
 * Copyright 2004-present Facebook. All Rights Reserved.
 *
 * @format
 */
'use strict';

const path = require('path');
const runner = require('../runner.js');

const TEST_PROJ_DIR = path.join(__dirname, './test-project');

test('./test-project', () => {
  console.log = jest.fn();
  const exitCode = runner.run('flow', [TEST_PROJ_DIR]);
  expect(console.log.mock.calls).toMatchSnapshot();
});
