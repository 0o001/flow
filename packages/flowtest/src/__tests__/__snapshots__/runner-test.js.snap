// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./test-project 1`] = `
Array [
  Array [
    "Using Flow v%s (%s)",
    "0.83.0",
    "/data/users/jmbrown/fbsource/fbcode/flow/packages/flowtest/node_modules/flow-bin/flow-linux64-v0.83.0/flow",
  ],
  Array [
    "Discovering flowtests in root \`%s\`",
    "/data/users/jmbrown/fbsource/fbcode/flow/packages/flowtest/src/__tests__/test-project",
  ],
  Array [
    "Found %d flowtest%s in root \`%s\`",
    6,
    "s",
    "/data/users/jmbrown/fbsource/fbcode/flow/packages/flowtest/src/__tests__/test-project",
  ],
  Array [
    "Using the existing flow server and running flow status",
  ],
  Array [
    "%s: %s",
    "[31m[1m[âœ—] FAIL[22m[39m",
    "/data/users/jmbrown/fbsource/fbcode/flow/packages/flowtest/src/__tests__/test-project/a/b/c/d/__flowtests__/fail.js",
  ],
  Array [
    "%s: %s",
    "[32m[1m[âœ“] PASS[22m[39m",
    "/data/users/jmbrown/fbsource/fbcode/flow/packages/flowtest/src/__tests__/test-project/a/b/c/d/__flowtests__/pass.js",
  ],
  Array [
    "%s: %s",
    "[31m[1m[âœ—] FAIL[22m[39m",
    "/data/users/jmbrown/fbsource/fbcode/flow/packages/flowtest/src/__tests__/test-project/a/b/c/d/fail-flowtest.js",
  ],
  Array [
    "%s: %s",
    "[32m[1m[âœ“] PASS[22m[39m",
    "/data/users/jmbrown/fbsource/fbcode/flow/packages/flowtest/src/__tests__/test-project/a/b/c/d/pass-flowtest.js",
  ],
  Array [
    "%s: %s",
    "[31m[1m[âœ—] FAIL[22m[39m",
    "/data/users/jmbrown/fbsource/fbcode/flow/packages/flowtest/src/__tests__/test-project/fail-flowtest.js",
  ],
  Array [
    "%s: %s",
    "[32m[1m[âœ“] PASS[22m[39m",
    "/data/users/jmbrown/fbsource/fbcode/flow/packages/flowtest/src/__tests__/test-project/pass-flowtest.js",
  ],
  Array [
    "%d tests passed, %d tests failed",
    3,
    3,
  ],
]
`;
